Makefile

```diff
@@ -27,7 +27,7 @@ static ?=       ## Enable static linking
 O := .build
 SOURCES := $(shell find src -name '*.cr')
 SPEC_SOURCES := $(shell find spec -name '*.cr')
-FLAGS := $(if $(release),--release )$(if $(stats),--stats )$(if $(progress),--progress )$(if $(threads),--threads $(threads) )$(if $(debug),-d )$(if $(static),--static )$(if $(LDFLAGS),--link-flags="$(LDFLAGS)" )
+FLAGS := $(if $(release),--release )$(if $(stats),--stats )$(if $(progress),--progress )$(if $(threads),--threads $(threads) )$(if $(debug),-d )$(if $(static),--static )$(if $(LDFLAGS),--link-flags="$(LDFLAGS)" ) $(EXTRA_FLAGS)
```

Comment:
Would it not to be better to just do something like
```make
override FLAGS += $(if $(release),--release )$(if $(stats),--stats )$(if $(progress),--progress )$(if $(threads),--threads $(threads) )$(if $(debug),-d )$(if $(static),--static )$(if $(LDFLAGS),--link-flags="$(LDFLAGS)" )
```
That way specifying `make all FLAGS=--totally-existant-flag` would just append `--totally-existant-flag` to flags. And you wouldn't have to type the extra `EXTRA_`

Reply:
Yeah thats what I meant, I should learn more makefile...

Reply:
> That way specifying `make all FLAGS=--totally-existant-flag` would just prepend `--totally-existant-flag` to the old FLAGS. And you wouldn't have to type the extra `EXTRA_`

That indeed an option, but it would not be backward compatible – now you can completely override `FLAGS`. If Crystal maintainers are okay with this change of behaviour, I’ll change it to the suggested variant.

Reply:
I had the same concern at first but then there are actually no default values in `FLAGS`, all depend on another variable being set. So previously specifying `FLAGS` and for example `release` didn't work, you already had to do `FLAGS=--release --whatever`, which will continue to work just the same. Only if you would have specified `release=1 FLAGS=--release` there would be a  slight change, but that was kinda invalid previously already.

Reply:
With my suggested change, `make all release=1 FLAGS=--no-debug` would build with the flags `--no-debug --release`

Reply:
@Jens0512 Is `override` really necessary? I think that `FLAGS += …` should work too…

Reply:
@jirutka Override is absolutely necessary